skbio/stats/distance/_cutils.pyx

```diff
@@ -293,3 +293,119 @@ def permanova_f_stat_sW_cy(TReal[:, ::1] distance_matrix,
             s_W += local_s_W/group_sizes[group_idx]
 
     return s_W
+
+from libc.math cimport isnan, sqrt, acos, fabs
+
+cimport numpy as cnp
+
+ctypedef fused floating:
```

Comment:
Should add a note saying that these types and functions are ported from hdmedians. Should indicate if you made any change to the code. Then copy the hdmedians license files to the `licenses` folder of skbio.